api/client/application/client_test.go

```diff
@@ -330,107 +353,131 @@ func (s *applicationSuite) TestDestroyApplications(c *gc.C) {
-	})
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	result := new(params.DestroyApplicationResults)
+	results := params.DestroyApplicationResults{}
+	args := params.DestroyApplicationsParams{
+		Applications: []params.DestroyApplicationParams{
+			{ApplicationTag: "application-foo"},
+		},
+	}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("DestroyApplication", args, result).SetArg(2, results).Return(errors.New("expected 1 result(s), got 0"))
+
+	client := application.NewClientFromCaller(mockFacadeCaller)
 	_, err := client.DestroyApplications(application.DestroyApplicationsParams{
 		Applications: []string{"foo"},
 	})
 	c.Assert(err, gc.ErrorMatches, `expected 1 result\(s\), got 0`)
 }
 
 func (s *applicationSuite) TestDestroyApplicationsInvalidIds(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
 	expectedResults := []params.DestroyApplicationResult{{
 		Error: &params.Error{Message: `application name "!" not valid`},
 	}, {
 		Info: &params.DestroyApplicationInfo{},
 	}}
-	client := newClient(func(objType string, version int, id, request string, a, response interface{}) error {
-		out := response.(*params.DestroyApplicationResults)
-		*out = params.DestroyApplicationResults{expectedResults[1:]}
-		return nil
-	})
-	results, err := client.DestroyApplications(application.DestroyApplicationsParams{
+	result := new(params.DestroyApplicationResults)
+	results := params.DestroyApplicationResults{expectedResults[1:]}
+
+	applications := []params.DestroyApplicationParams{
+		{ApplicationTag: names.NewApplicationTag("foo").String()},
+	}
+	args := params.DestroyApplicationsParams{Applications: applications}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("DestroyApplication", args, result).SetArg(2, results).Return(nil)
+	client := application.NewClientFromCaller(mockFacadeCaller)
+	res, err := client.DestroyApplications(application.DestroyApplicationsParams{
 		Applications: []string{"!", "foo"},
 	})
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(results, jc.DeepEquals, expectedResults)
+	c.Assert(res, jc.DeepEquals, expectedResults)
 }
 
 func (s *applicationSuite) TestDestroyConsumedApplicationsArity(c *gc.C) {
-	client := newClient(func(objType string, version int, id, request string, a, response interface{}) error {
-		return nil
-	})
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	result := new(params.ErrorResults)
+	force := false
+	args := params.DestroyConsumedApplicationsParams{
+		Applications: []params.DestroyConsumedApplicationParams{
+			{ApplicationTag: names.NewApplicationTag("foo").String(), Force: &force},
+		},
+	}
 	destroyParams := application.DestroyConsumedApplicationParams{
 		[]string{"foo"}, false, nil,
 	}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("DestroyConsumedApplications", args, result).Return(errors.New("expected 1 result(s), got 0"))
```

Comment:
The "expected 1 results(s), not 0" does result from the mock result, it should come from api code when the number of request does not equal the number of results received, see line 586.